cdk/apps/dracarys/dracarys/dracarys_stack.py

```diff
@@ -0,0 +1,61 @@
+from aws_cdk import (
+    aws_lambda,
+    aws_iam,
+    aws_s3,
+    aws_sqs,
+    aws_ssm,
+    aws_lambda_event_sources as lambda_event_source,
+    Stack,
+    Duration
+)
+
+from constructs import Construct
+
+class DracarysStack(Stack):
+
+    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        lambda_role = aws_iam.Role(scope=self, id='dracarys-lambda-role',
+                                assumed_by =aws_iam.ServicePrincipal('lambda.amazonaws.com'),
+                                role_name='dracarys-lambda-role',
+                                managed_policies=[
+                                aws_iam.ManagedPolicy.from_aws_managed_policy_name(
+                                    'service-role/AWSLambdaBasicExecutionRole'),
+                                aws_iam.ManagedPolicy.from_aws_managed_policy_name(
+                                    'service-role/AWSLambdaVPCAccessExecutionRole'),
+                                aws_iam.ManagedPolicy.from_aws_managed_policy_name(
+                                    'SecretsManagerReadWrite'),
+                                aws_iam.ManagedPolicy.from_aws_managed_policy_name(
+                                    'AmazonS3FullAccess')
+                                ])
+
+        # Fetch dracarys queue ARN from pre-existing SSM
+        # created by terraform's data portal pipeline stack
+        queue_name = "/data_portal/backend/sqs_dracarys_queue"
+        queue_arn = aws_ssm.StringParameter.from_string_parameter_attributes(self, "dracarys_sqs_queue",
+                parameter_name=queue_name).string_value
+
+        queue = aws_sqs.Queue.from_queue_arn(self, id="data-portal-dracarys-queue", queue_arn=queue_arn)
+
+        bucket = aws_s3.Bucket(self, "umccr-datalake-dev")
```

Comment:
I suppose the bucket name would be parameterised in the future?

Reply:
What's the current best practice? Lambda env? SSM? I've seen this: https://github.com/umccr/infrastructure/blob/master/cdk/apps/thebus/stacks/thebus_stack.py#L35 and made me wonder 🤔

Reply:
I think according to the best practices [guide](https://blog.serverlessadvocate.com/serverless-aws-cdk-pipeline-best-practices-patterns-part-1-ab80962f109d) it should be neither, but a stack specific configs.

Reply:
Went through that 2-part guide this morning (such a shame part 3 is not there yet since it deals with variables/secrets :__/). I think that following that CDK Pipelines requires requires quite some time to refactor/rework from here, what "best" practice would you think it'd be acceptable to adopt **today**?

Reply:
> just passing props from some parent stack?

I found the parsing of properties between stacks much easier with the TypeScript implementation of CDK (if that's of any consolation). For me it was enough to switch (which I did for the cttso-ica-to-pieriandx repo).  

I've also just got a bash script which I use to sync ssm parameters with (https://github.com/umccr/cttso-ica-to-pieriandx/blob/main/deploy/cttso-ica-to-pieriandx-cdk/scripts/update-params.sh) but this is probably better handled with terraform

Reply:
> What's the current best practice across our stacks? Lambda env? SSM? I've seen this: [master/cdk/apps/thebus/stacks/thebus_stack.py#L35](https://github.com/umccr/infrastructure/blob/master/cdk/apps/thebus/stacks/thebus_stack.py?rgh-link-date=2023-03-20T00%3A19%3A30Z#L35) and made me wonder 🤔

I have a constants file I use to import variables like this

https://github.com/umccr/cttso-ica-to-pieriandx/blob/main/deploy/cttso-ica-to-pieriandx-cdk/constants.ts 

Similar to holmes (@andrewpatto)

Reply:
Cool, thanks Alexis, I think I might just go for the constants route for now, thanks for chiming in!